### PR TITLE
Add email validation

### LINKED ISSUE #28451: Validate email before sending out public link from dialog
### Steps

1. Create a public Link
1. Enter a wrong Mail Address (e.g. without @)
1. Click save

### Expected

Link not saved, error message, dialog still there.

### Actual

Dialog reappears four times.

### Version
10.0.1

@felixheidecke can you have a look ?

### PR DESCRIPTION
**Pull request by lordelix**
- Add email validation for public link creation
- show email field even if the model !isNew

## Related Issue

Fixes https://github.com/owncloud/core/issues/28451

## How was this tested
- Manually tested (visual test) on latest Firefox, Chrome on MacOS.
- Manually tested (visual test) on latest Firefox, Chrome, IE1 and Edge on Windows.
- Changed and added to Jasmine test.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by felixheidecke** [comment]
I'm still improving. Pls no review yet ;-)

**Comment by felixheidecke** [comment]
@PVince81 pls have a look.

**Comment by PVince81** [comment]
@felixheidecke can you list the test cases you tested manually ? (it's the "How was this tested" section in the PR template)

Code looks good

**Comment by felixheidecke** [comment]
@PVince81 updated and added tests :-)

**Review by PVince81 — APPROVED** [review]
:+1:

**Comment by PVince81** [comment]
Forget about Jenkins... git fetch failures...

I ran the JS tests locally and they all passed, merging.

@felixheidecke please backport to stable10

**Comment by DeepDiver1975** [comment]
this got lost because no labels and no milestone was set ... damn ... is it that hard?

**Comment by PVince81** [comment]
@felixheidecke please backport as soon as possible

File: core/js/sharedialogmailview.js [hunk 1]
```diff
@@ -55,6 +55,10 @@
 			var itemType = this.itemModel.get('itemType');
 			var itemSource = this.itemModel.get('itemSource');
 
+			if (!this.validateEmail(recipientEmail)) {
+				deferred.reject();
```
Comment (PVince81) [inline core/js/sharedialogmailview.js]: I think we should return here to avoid sending the ajax request below

**Pull request merged**